lib/patch-https-request-for-proxying.js

```diff
@@ -0,0 +1,98 @@
+/*
+ * Monkey-patch https.request to proxy request to the broker server
+ * according the the env variables `https_proxy` and `no_proxy`:
+ * get the https proxy from the env. proxy requests unless the
+ * broker server is in the no_proxy list.
+ * Start from the "Entry point" below.
+ */
+const url = require('url');
+const tunnel = require('tunnel');
+const {brokerServerUrl} = require('./config');
+const brokerServer = url.parse(brokerServerUrl || '');
+brokerServer.port = brokerServer.port ||
+  brokerServer.protocol === 'https:' ? '443' : '80';
+const httpsProxy = process.env.HTTPS_PROXY || process.env.https_proxy;
```

Comment:
(and allow people to use `.env` predictably).

I know that config also modifies process.env, but I think relying on that side-effect is a step too far.